presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleTester.java

```diff
@@ -123,12 +124,24 @@ public ConnectorId getCurrentConnectorId()
                 return empty();
             }
 
-            return getOnlyElement(matching).apply(node, captures, context);
+            return apply(node, getOnlyElement(matching), context);
```

Comment:
I would rather get rid of `RuleSetAdapter` because:
* it assumes that rules in `RuleSet` have particular semantics (e.g: only one rule matches)
* it wraps and executes rules in a different way than during actual optimization
* it doesn't allow to test how rules interact (e.g: trait rules before other rewrite rules)
* it only allows to test rule set which are grouped in `RuleSet`, but we might want to test not grouped rules (e.g: trait rules and rewrites).
* it is a work around `RuleAssert` limitation of testing singular rule. I would rather test multiple rules explicitly in `RuleAssert` with assertions.

Reply:
Please file a ticket for that. 

The thing that some is not perfect does not mean it should not be used, maintained or improved. Do you have something I could use instead? Otherwise, your comment is out of scope.